### PR TITLE
Remove animation of chart on data update & fix hydration issue on login page

### PR DESCRIPTION
**Pull request by ThiloAschebrock**
- [Fix hydration issue on login page](https://github.com/TNG/next-generation-scrum-poker/commit/9f5e7c6ec5284451a3824c4550e5d962f38d4dff) 

- [Disable pie-chart animation on update](https://github.com/TNG/next-generation-scrum-poker/commit/f4644c32743f221f3e43e9c937f1c0b0d270bfbf)

### CODE REVIEW

**Comment by ThiloAschebrock** [comment]
Issue: Animation is canceled when websocket data arrives. Solution should be to use deep-compare.

File: frontend/src/helpers/helpers.ts [hunk 1]
```diff
@@ -1,3 +1,18 @@
...
+export function useDeepCompareMemoize<T>(value: T): T {
+  const ref = useRef<T>(value);
+  const signalRef = useRef<number>(0);
+
+  if (!dequal(value, ref.current)) {
+    ref.current = value;
+    signalRef.current += 1;
+  }
+
+  return useMemo(() => ref.current, [signalRef.current]);
```
Comment (lukastaegert) [inline frontend/src/helpers/helpers.ts:17]: While this is definitely an intriguing technique, I also find it slightly hard to follow and a rather unusual use for `useMemo`. Especially since `ref` and `signalRef` only change together, so I would expect the `useMemo` to not make a difference?
Couldn't we just directly use the `ref`, i.e.

```js
export function useDeepCompareMemoize<T>(value: T): T {
   const ref = useRef<T>(value);

   if (!dequal(value, ref.current)) {
     ref.current = value;
   }

   return ref.current;
}
```

or am I overlooking something here?

File: frontend/src/helpers/helpers.ts [hunk 2]
```diff
@@ -1,3 +1,18 @@
+import { dequal } from 'dequal';
```
Comment (lukastaegert) [inline frontend/src/helpers/helpers.ts:1]: Considering we are only using it for JSON data, can we also use `dequal/lite`?

Comment (lukastaegert) [reply frontend/src/helpers/helpers.ts:1]: BTW dequal seems like a rather nice package I did not know yet 👍

Comment (ThiloAschebrock) [reply frontend/src/helpers/helpers.ts:1]: Using `dequal/lite` now.

File: frontend/src/helpers/helpers.ts (hunk 1 above)
Comment (ThiloAschebrock) [reply frontend/src/helpers/helpers.ts]: Maybe we are, but I don't see it either. I was inspired by the implementation of https://github.com/kentcdodds/use-deep-compare-effect/blob/main/src/index.ts#L32 without questioning the actual implementation. Going by the history of this implementation, the `signalRef` was first introduced as a speedup, but later on the return value also change. I would question if the speedup aspect is actually still present.

In order to keep it simple here, I have gone with the standard implementation that you suggested. Moreover, I added test coverage.

**Review by lukastaegert — APPROVED** [review]
Works perfectly!
